### PR TITLE
Enhancement: Improve support for web app deployment (tests written by @coi-gov-pl guys)

### PR DESCRIPTION
**Pull request by Cosaquee**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/coi-gov-pl/puppet-jboss/pull/78" target="_blank">ReviewNinja

File: manifests/deploy.pp [hunk 1]
```diff
@@ -27,10 +27,15 @@
   $servergroups = hiera('jboss::deploy::servergroups', undef),
   $controller   = $::jboss::controller,
   $runasdomain  = $::jboss::runasdomain,
+  $runtime_name = undef,
 ) {
   include jboss
   include jboss::internal::runtime::node
 
+  if $runtime_name != undef {
+    validate_re($runtime_name, '.+.(\.ear|\.zip|\.war|\.jar)$')
```
Comment (cardil) [inline manifests/deploy.pp]: !fix Should be `.+(\.ear|\.zip|\.war|\.jar)$`. Also add a fail message for end-users

File: tests/examples/deploy.pp [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class { 'jboss':
```
Comment (cardil) [inline tests/examples/deploy.pp]: !fix Remove this example file, and update the working smoke test `tests/deploy.pp` to utilize new `runtime_name` parameter

File: spec/unit/provider/jboss_deploy/jboss_deploy_spec.rb [hunk 3]
```diff
@@ -292,5 +292,27 @@
       it { expect(subject).to eq(["super-crm", "super-crm-1"]) }
     end
 
+    describe '#refresh' do
```
Comment (cardil) [inline spec/unit/provider/jboss_deploy/jboss_deploy_spec.rb]: !resolve add test for runtime_name, also

File: lib/puppet_x/coi/jboss/factsrefresher.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        fct = Facter.fact value
+        fct.instance_variable_set(:@value, {})
+    end
+
+    private
+    # Method used to validate if fact is system or module fact
+    # @param {String} name of the fact to be validated
+    # @return {true} if fact name is from jboss module
+    def validate_fact_name value
+      value.to_s.start_with? 'jboss'
```
Comment (cardil) [inline lib/puppet_x/coi/jboss/factsrefresher.rb]: !fix Change to `'jboss_'` for more explicit

File: lib/puppet_x/coi/jboss/factsrefresher.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    # Method used to delete resolves in given fact
+    # @param {String} fact name that resolves should be deleted
+    def delete_resolves value
+        fct = Facter.fact value
+        fct.instance_variable_set(:@resolves, [])
+    end
+
+    # Method used to delete values in given fact
+    # @param {String} fact name that value should be deleted
+    def delete_value value
```
Comment (cardil) [inline lib/puppet_x/coi/jboss/factsrefresher.rb]: !fix change to `key` or `symbol`

File: lib/puppet_x/coi/jboss/factsrefresher.rb [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+class Puppet_X::Coi::Jboss::FactsRefresher
+  class << self
+
+    # Method to refresh given facts
+    # @param {[Symbol]}  list of fact symbols to be refreshed
+    def refresh_facts value
+      facts = Facter.list # get list of symbols of facts in the system
+      value.each do |val|
+        raise Puppet::Error, 'You can only delete fact that are made by jboss_module(start with jboss_)' unless validate_fact_name val
+          delete_resolves val
```
Comment (cardil) [inline lib/puppet_x/coi/jboss/factsrefresher.rb]: !fix bad ident

File: lib/puppet/type/jboss_deploy.rb [hunk 7]
```diff
@@ -52,5 +60,9 @@
     desc "Retry timeout in seconds"
     defaultto 1
   end
-  
+
+  def refresh
```
Comment (Cosaquee) [inline lib/puppet/type/jboss_deploy.rb]: docs

File: spec/unit/provider/jboss_deploy/jboss_deploy_spec.rb (hunk 3 above)
Comment (Cosaquee) [inline spec/unit/provider/jboss_deploy/jboss_deploy_spec.rb]: !fixed

File: tests/examples/deploy.pp (hunk 2 above)
Comment (Cosaquee) [inline tests/examples/deploy.pp]: !fixed

File: lib/puppet_x/coi/jboss/factsrefresher.rb (hunk 6 above)
Comment (Cosaquee) [inline lib/puppet_x/coi/jboss/factsrefresher.rb]: !fixed

File: lib/puppet_x/coi/jboss/factsrefresher.rb (hunk 5 above)
Comment (Cosaquee) [inline lib/puppet_x/coi/jboss/factsrefresher.rb]: !fixed

File: lib/puppet_x/coi/jboss/factsrefresher.rb (hunk 4 above)
Comment (Cosaquee) [inline lib/puppet_x/coi/jboss/factsrefresher.rb]: !fixed

File: manifests/deploy.pp (hunk 1 above)
Comment (Cosaquee) [inline manifests/deploy.pp]: !fixed

**Comment by cardil** [comment]
:+1:

**Comment by frodoslaw** [comment]
:+1:

**Comment by cardil** [comment]
Green bar :star2:

**Pull request merged**